R/GeoLift.R

```diff
@@ -2482,17 +2486,17 @@ GeoLiftMarketSelection <- function(data,
                             Average_MDE = lift,
                             Power = power)
 
-  # Step 10 - Remove tests out of budget (if aplicable)
-  if (!is.null(budget)){
-    resultsM <- resultsM %>% dplyr::filter(budget > Investment)
-  }
-
-  # Step 11: Adjust signs if Negative Lift
+  # Step 10: Adjust signs if Negative Lift
   if (min(effect_size) < 0){
     resultsM$Investment <- -1*resultsM$Investment
     results$Investment <- -1*results$Investment
   }
 
+  # Step 11 - Remove tests out of budget (if aplicable)
+  if (!is.null(budget)){
+    resultsM <- resultsM %>% dplyr::filter(budget > Investment)
```

Comment:
Can we change this to absolute values? Just in case the user decides to input a negative value for budget (if they run a negative lift?).  That could catch that fringe case and avoid filtering everything out.  Something like `... dplyr::filter(abs(budget) > abs(Investment))`

Reply:
Good idea, I'll make the change especially since we don't specify that budget must be positive in the function's description.

Reply:
Cool, since we are multiplying resultsM$Investment by -1 in the previous step as well, shouldn't that also be abs(Invemstent)?